src/components/share-modal.js

```diff
@@ -1,61 +1,110 @@
 import { useSelector } from "react-redux";
 import { get_friends_for_author } from "../api/follower_api";
 import { useEffect, useState } from "react";
-import { Box, Modal, Table } from "@mui/material";
+import {
+  Alert,
+  Box,
+  Button,
+  Input,
+  Modal,
+  Snackbar,
+  Typography,
+} from "@mui/material";
 import { post_inbox } from "../api/inbox_api";
+import "./share-modal.css";
 
 const ShareModal = (props) => {
   const user = useSelector((state) => state.user);
   const [friends, setFriends] = useState(null);
   const [filter, setFilter] = useState("");
 
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarServerity, setSnackbarServerity] = useState("");
+
   const style = {
     position: "absolute",
     top: "50%",
     left: "50%",
     transform: "translate(-50%, -50%)",
     width: 400,
     bgcolor: "background.paper",
-    boxShadow: 24,
+    boxShadow: 20,
     p: 4,
   };
 
   useEffect(() => {
     get_friends_for_author(user.id, setFriends);
   }, []);
 
-  const shareToInbox = (authorId, post) => {
-    post_inbox(authorId, post);
-  };
-
   const filterFriend = (friend) => {
     return friend.displayName.toLowerCase().includes(filter.toLowerCase());
   };
 
+  const handleClose = () => {
+    setSnackbarOpen(false);
+  };
+
+  const inbox_result = (status) => {
+    if (status === 202) {
+      setSnackbarServerity("success");
+    }
+    if (status === 409) {
+      setSnackbarServerity("error");
+    }
+    setSnackbarOpen(true);
+  };
+
   return (
-    friends && (
-      <Modal {...props}>
-        <Box sx={style}>
-          <input
-            name="filter"
-            type="text"
-            placeholder="Search friend"
-            onChange={(e) => setFilter(e.target.value)}
-            value={filter}
-          />
-          {friends.items
-            .filter((friend) => filterFriend(friend))
-            .map((friend) => (
-              <div>
-                {friend.displayName}
-                <button onClick={() => shareToInbox(friend.id, props.post)}>
-                  Send
-                </button>
-              </div>
-            ))}
-        </Box>
-      </Modal>
-    )
+    <div>
+      {friends && (
+        <Modal {...props}>
+          <Box sx={style}>
+            <Input
+              sx={{ width: "100%", mb: 3 }}
+              name="filter"
+              type="text"
+              placeholder="Search friend"
+              onChange={(e) => setFilter(e.target.value)}
+              value={filter}
+            />
+            {friends.items
+              .filter((friend) => filterFriend(friend))
+              .map((friend) => (
+                <div className="friend-row">
+                  <img id="profile-image" src={friend.profileImage} />
```

Comment:
Needs a check to see if profile image is blank so that we can use the default profile photo is used